test/common/config/subscription_impl_test.cc

```diff
@@ -92,57 +93,57 @@ INSTANTIATE_TEST_SUITE_P(SubscriptionImplTest, SubscriptionImplInitFetchTimeoutT
 }
 
 // Validate that multiple streamed updates succeed.
 TEST_P(SubscriptionImplTest, ResponseStream) {
   startSubscription({"cluster0", "cluster1"});
-  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", true);
-  EXPECT_TRUE(statsAre(2, 1, 0, 0, 0, TEST_TIME_MILLIS, 7148434200721666028));
+  EXPECT_TRUE(statsAre(2, 1, 0, 0, 0, TEST_TIME_MILLIS, 7148434200721666028, "0"));
   deliverConfigUpdate({"cluster0", "cluster1"}, "1", true);
-  EXPECT_TRUE(statsAre(3, 2, 0, 0, 0, TEST_TIME_MILLIS, 13237225503670494420U));
+  EXPECT_TRUE(statsAre(3, 2, 0, 0, 0, TEST_TIME_MILLIS, 13237225503670494420U, "1"));
 }
 
 // Validate that the client can reject a config.
 TEST_P(SubscriptionImplTest, RejectConfig) {
   startSubscription({"cluster0", "cluster1"});
-  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", false);
-  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0, ""));
 }
 
 // Validate that the client can reject a config and accept the same config later.
 TEST_P(SubscriptionImplTest, RejectAcceptConfig) {
   startSubscription({"cluster0", "cluster1"});
-  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", false);
-  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", true);
-  EXPECT_TRUE(statsAre(3, 1, 1, 0, 0, TEST_TIME_MILLIS, 7148434200721666028));
+  EXPECT_TRUE(statsAre(3, 1, 1, 0, 0, TEST_TIME_MILLIS, 7148434200721666028, "0"));
 }
 
 // Validate that the client can reject a config and accept another config later.
 TEST_P(SubscriptionImplTest, RejectAcceptNextConfig) {
   startSubscription({"cluster0", "cluster1"});
-  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", false);
-  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(2, 0, 1, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "1", true);
-  EXPECT_TRUE(statsAre(3, 1, 1, 0, 0, TEST_TIME_MILLIS, 13237225503670494420U));
+  EXPECT_TRUE(statsAre(3, 1, 1, 0, 0, TEST_TIME_MILLIS, 13237225503670494420U, "1"));
 }
 
 // Validate that stream updates send a message with the updated resources.
 TEST_P(SubscriptionImplTest, UpdateResources) {
   startSubscription({"cluster0", "cluster1"});
-  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0));
+  EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
   deliverConfigUpdate({"cluster0", "cluster1"}, "0", true);
-  EXPECT_TRUE(statsAre(2, 1, 0, 0, 0, TEST_TIME_MILLIS, 7148434200721666028));
+  EXPECT_TRUE(statsAre(2, 1, 0, 0, 0, TEST_TIME_MILLIS, 7148434200721666028, "0"));
```

Comment:
Maybe some of these can be slightly more interesting version identifiers..

Reply:
sgtm. What's a good example of more interesting version identifier? 
Is "1.2.3.4" or "a.b.c.d" fairly representative?

Reply:
I think anything that proves we are retaining string data; how about "v25-ubuntu18-beta" or similar?

Reply:
Done. For me "0" and "1" are sufficiently entertaining, but I've tried to spruce up some tests. We can't have too many "42".